pkg/loki/modules.go

```diff
@@ -1053,6 +1053,10 @@ func (t *Loki) initQueryFrontend() (_ services.Service, err error) {
 		t.Server.HTTP.Path("/api/prom/tail").Methods("GET", "POST").Handler(defaultHandler)
 	}
 
+  if t.Cfg.Frontend.ExperimentalAPIsEnabled {
```

Comment:
Not sure if we need a flag if this is an experimental API

Reply:
I like the idea of just not responding to these endpoints at all when turned off. Makes it very quick to stop the bleeding if there's something terribly wrong.